### ISSUE TITLE
Use http2 for Kubernetes Client Cache / Reflector

### ISSUE THREAD

**Issue by holgerkoser** [gardener/dashboard#656]
**What would you like to be added**:

* Use http2 for Kubernetes Client Cache / Reflector. This allows session heartbeat and apiserver stickyness.
* Use session heartbeat for setting the store state to STALE or FRESH.
* For Reflector test use a TestStore impl instead of fragile nextTick assumptions.
* Show error page if cache is stale (maybe not related to this issue)

**Why is this needed**:

We have multiple apiserver instances where each instance has it's own watch-cache. If we do an initial list to fill the store we also receive a `resourceVersion` which we use for the following watch. The received `resourceVersion` is different for each apiserver instance. In the following example we have 6 virtual apiserver pods.
```bash
$ kubectl get -n garden pods --selector=app=virtual-garden,component=kube-apiserver
NAME                                             READY   STATUS    RESTARTS   AGE
virtual-garden-kube-apiserver-69b995c65f-47hzp   2/2     Running   1          43h
virtual-garden-kube-apiserver-69b995c65f-crv67   2/2     Running   1          19h
virtual-garden-kube-apiserver-69b995c65f-cwmjm   2/2     Running   1          43h
virtual-garden-kube-apiserver-69b995c65f-hf4l2   2/2     Running   1          169m
virtual-garden-kube-apiserver-69b995c65f-n65p6   2/2     Running   1          169m
virtual-garden-kube-apiserver-69b995c65f-ndffw   2/2     Running   1          43h
```
If we e.g. list list all quotas we receive 4 different `resourceVersion` values `1401401153`, `1401401238`, `1401401374` and `1401401405` depending on the instance which responded to the http request. 
```bash
$ for i in `seq 10`; do kubectl get --raw "/apis/core.gardener.cloud/v1beta1/quotas?resourceVersion=0&limit=500" | jq -r .metadata.resourceVersion; done;
1401401153
1401401238
1401401405
1401401238
1401401405
1401401374
1401401238
1401401153
1401401374
1401401238
```
If you now start a watch let say with resourceVersion `1401401153` we could be directed to an instance for which this resource version is too old.
```bash
$ kubectl get --raw "/apis/core.gardener.cloud/v1beta1/quotas?watch=true&resourceVersion=1401401153"
```
In this case we get a StatusError event with 410 Gone and the message "too old resource version: 1401401153 (1401401238)".
```json
{
  "type": "ERROR",
  "object": {
    "kind": "Status",
    "apiVersion": "v1",
    "metadata": {},
    "status": "Failure",
    "message": "too old resource version: 1401401153 (1401401238)",
    "reason": "Gone",
    "code": 410
  }
}
```
Using http2 session solves this problem.

**Comment by holgerkoser** [comment]
Closed by #972

**Comment by holgerkoser** [comment]
Closed by #972
